### PR TITLE
Enter/Exit screener signal

### PR DESCRIPTION
**Pull request by IvanIvanoff**
## Changes
- Implement `screener_signal` which notifies what projects exited or entered the screener.
- In order not to mark all projects as new, when the screener is created or updated, a `post_create_process/1` or `post_update_process/2` function is called. They properly set the state, so only changes in screener are made into signals.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/map_utils.ex [hunk 1]
```diff
@@ -89,10 +89,8 @@ defmodule Sanbase.MapUtils do
   # Private functions
 
   @compile {:inline, atomize: 1}
-  defp atomize(atom) when is_atom(atom), do: atom
-
-  defp atomize(str) when is_binary(str),
-    do: str |> Inflex.underscore() |> String.to_existing_atom()
+  defp atomize(value) when is_atom(value) or is_binary(value),
```
Comment (tspenov) [inline lib/map_utils.ex:92]: Why is this needed ? do we have atoms in camel case ?

File: lib/sanbase/model/project/list/list_selector.ex [hunk 2]
```diff
@@ -1,8 +1,67 @@
...
+        aggregation: spec(is_atom())
+      })
+
+    Enum.map(filters, &conform(&1, filter_schema))
+    |> Enum.find(&match?({:error, _}, &1))
+    |> case do
+      nil ->
+        Enum.map(
+          filters,
+          &([:metric, :from, :to, :operator, :threshold, :aggregation] -- Map.keys(&1))
```
Comment (tspenov) [inline lib/sanbase/model/project/list/list_selector.ex:52]: That is because you can't specify required keys in the schema, right ?

File: lib/sanbase/signals/evaluator/scheduler.ex [hunk 3]
```diff
@@ -60,6 +60,8 @@ defmodule Sanbase.Signal.Scheduler do
 
   # returns a tuple {updated_user_triggers, send_result_list}
   defp send_and_mark_as_sent(triggers) do
+    IO.inspect("SEND AND MARK")
```
Comment (tspenov) [inline lib/sanbase/signals/evaluator/scheduler.ex:63]: inspect left

File: lib/sanbase/signals/evaluator/scheduler.ex [hunk 4]
```diff
@@ -74,6 +76,10 @@ defmodule Sanbase.Signal.Scheduler do
           list when is_list(list) ->
             {:ok, updated_user_trigger} = update_last_triggered(user_trigger, list)
 
+            IO.inspect(user_trigger.trigger.settings.state,
```
Comment (tspenov) [inline lib/sanbase/signals/evaluator/scheduler.ex:79]: inspect left

File: lib/sanbase/signals/trigger/settings/screener_trigger_settings.ex [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+      kv = %{
+        type: ScreenerTriggerSettings.type(),
+        operation: settings.operation,
+        added_slugs: added_slugs,
+        removed_slugs: removed_slugs
+      }
+
+      template = """
+      🔔Projects' changes in the screener "Title goes here".
```
Comment (tspenov) [inline lib/sanbase/signals/trigger/settings/screener_trigger_settings.ex:124]: I think there should be also a link to the screener  :?

File: lib/sanbase/signals/trigger/validation/operation_validation.ex [hunk 6]
```diff
@@ -90,6 +90,14 @@ defmodule Sanbase.Signal.Validation.Operation do
   def valid_operation?(%{amount_up: value}) when is_number(value), do: :ok
   def valid_operation?(%{amount_down: value}) when is_number(value), do: :ok
 
+  # Validate screener signal
+  def valid_operation?(%{selector: _} = selector) do
+    case Sanbase.Model.Project.ListSelector.valid_selector?(selector) do
+      true -> :ok
+      false -> {:error, "The provided selector is not correct."}
```
Comment (tspenov) [inline lib/sanbase/signals/trigger/validation/operation_validation.ex:97]: `correct` -> `valid` ?

File: lib/map_utils.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/map_utils.ex:92]: When we have JSON in the api in a free format (the `:json` custom type) we cannot enforce the casing. In all other parts of the API there's a built-in transformer which converts camel case to underscore, so the frontend uses `allProjects` but we see it as `all_projects`.

File: lib/sanbase/model/project/list/list_selector.ex (hunk 2 above)
Comment (IvanIvanoff) [reply lib/sanbase/model/project/list/list_selector.ex:52]: yes, turned out this library is not supporting that.

File: lib/sanbase/signals/trigger/settings/screener_trigger_settings.ex (hunk 5 above)
Comment (IvanIvanoff) [reply lib/sanbase/signals/trigger/settings/screener_trigger_settings.ex]: This is automatically added: https://github.com/santiment/sanbase2/blob/master/lib/sanbase/signals/signal.ex#L126-L131
